azkaban-web-server/src/main/java/azkaban/webapp/WebMetrics.java

```diff
@@ -42,13 +44,24 @@
 
   WebMetrics() {
     registry = MetricsManager.INSTANCE.getRegistry();
-    setupAllMetrics();
+    setupStaticMetrics();
   }
 
-  private void setupAllMetrics() {
+  private void setupStaticMetrics() {
     webGetCall = MetricsUtility.addMeter("Web-Get-Call-Meter", registry);
     webPostCall = MetricsUtility.addMeter("Web-Post-Call-Meter", registry);
-    MetricsUtility.addLongGauge("fetchLogLatency", logFetchLatency, registry);
+    MetricsUtility.addGauge("fetchLogLatency", registry, logFetchLatency::get);
+  }
+
+  public void addExecutorManagerMetrics(ExecutorManager executorManager) {
+    MetricsUtility.addGauge("WEB-NumQueuedFlows", registry, executorManager::getQueuedFlowSize);
+
+    /**
+     * TODO: Currently {@link ExecutorManager#getRunningFlows()} includes both running and queueing jobs.
+     * We need to differentiate it in future.
+     */
+    Supplier<Long> supplier = () -> executorManager.getRunningFlows().size() - executorManager.getQueuedFlowSize();
```

Comment:
Good question. I guess we should make something synchronized, in case List<RunningFLow> or List<QueuedFlows> changes. Yet it can not solve this problem if we mark this supplier function to synchronized, right? That should not help, because RunningFLow or QueuedFlows may vary when we call this supplier function.